File: Array/_27_Remove_Elements/python/RemoveElements.py
 # T.C.: O(n)
 # S.C.: O(1)
from typing import List


class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeElements(self, nums: List[int], val: int) -> int:
        valid_size = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[valid_size] = nums[i]
                valid_size +=1
        return valid_size        
                
if __name__ == "__main__":
   print(Solution().removeElements([0,1,2,2,3,0,4,2],2))

File: Array/_27_Remove_Elements/java/RemoveElements.java
public class RemoveElements {
    public static void main(String[] args) {
       System.out.println(removeElements(new int[] {0,1,2,2,3,0,4,2}, 2));
    }
    static int removeElements(int[] nums, int val){
        if(nums.length == 0) {
            return 0;
        }
        int validSize = 0;
        for (int i = 0; i < nums.length; i++) {
            if(nums[i] != val) {
                nums[validSize] = nums[i];
                validSize++;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            System.out.print(nums[i] +" ");
        }
        return validSize;
    }
}